src/controller/Controller.spec.js

```diff
@@ -73,4 +73,45 @@ describe('controller', () => {
       done();
     }, 10);
   });
+
+  it('Should call up function on blur if the key was down', done => {
+    const controller = new Controller([
+      {
+        key: 'Enter',
+        up: function() {
+          done();
+        }.bind(this)
+      }
+    ]);
+
+    const eventDown = newEvent('keydown');
+    eventDown.key = 'Enter';
+    window.dispatchEvent(eventDown);
+
+    const blurEvent = newEvent('blur');
+    window.dispatchEvent(blurEvent);
+
+    controller.update();
+  });
+
+  it('Should not call up function on blur if the key was not down', done => {
+    const controller = new Controller([
```

Comment:
Another potentially cleaner way to do this test would be to use something `sinon` and create a [fake](https://sinonjs.org/releases/v8.0.4/fakes/) as the `up` listener. You can then assert if the function was called or not:

```
expect(upFake.callCount).to.equal(0);
```

and that way you can avoid using a timeout, which might give undefined behavior (especially if it's potentially called twice in a failing case).